lib/spack/spack/test/cmd/env.py

```diff
@@ -3026,6 +3026,34 @@ def test_stack_view_activate_from_default(
         assert "FOOBAR=mpileaks" in shell
 
 
+def test_envvar_set_in_activate(tmpdir, mock_fetch, mock_packages, mock_archive, install_mockery):
+    filename = str(tmpdir.join("spack.yaml"))
+    with open(filename, "w", encoding="utf-8") as f:
+        f.write(
+            """\
+spack:
+  specs:
+    - cmake%gcc
+  env_vars:
+    set:
+      ENVAR_SET_IN_ENV_LOAD: "True"
+"""
+        )
+    with tmpdir.as_cwd():
+        env("create", "test", "./spack.yaml")
+        with ev.read("test"):
+            install()
+
+        test_env = ev.read("test")
+
+        output = env("activate", "--sh", "test")
+        assert "ENVAR_SET_IN_ENV_LOAD=True" in output
+
+        with test_env:
+            output = env("deactivate", "--sh")
+            assert "ENVAR_SET_IN_ENV_LOAD" not in output
```

Comment:
@tgamblin Thanks for pointing that out.  I think it's testing the right thing now.  Does this look right?

Reply:
No, because `"ENVAR_SET_IN_ENV_LOAD"` should be *in* the output when you deactivate.  If it's set on `activate`, it still needs to be unset on `deactivate`, so `spack env deactivate --sh`  should be printing something to do that. So this is weird